src/impl/peerconnection.cpp

```diff
@@ -1218,4 +1170,39 @@ void PeerConnection::resetCallbacks() {
 	gatheringStateChangeCallback = nullptr;
 }
 
+void PeerConnection::updateTrackSsrcCache(const Description &description) {
+	std::lock_guard lock(mTracksMutex); // read-only (writing to mTracksBySsrc)
```

Comment:
You should remove `read-only` from the comment. Also, you could use `unique_lock` instead of `lock_guard` for clarity.